### PR TITLE
invalidate NSURLSession to avoid memory leaks

### PR DESCRIPTION
**Pull request by derrickstaten**
@aaustin

### CODE REVIEW

**Comment by aaustin** [comment]
:+1:

File: Branch-SDK/Branch-SDK/BNCServerInterface.m [hunk 1]
```diff
@@ -121,6 +121,7 @@ - (void)genericHTTPRequest:(NSURLRequest *)request retryNumber:(NSInteger)retryN
         }
     }];
     [task resume];
+    [session finishTasksAndInvalidate];
```
Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/BNCServerInterface.m]: Should this be done in the completion block?

Comment (derrickstaten) [inline Branch-SDK/Branch-SDK/BNCServerInterface.m]: It doesn't appear to be necessary, per the [documentation](https://developer.apple.com/library/ios/documentation/Foundation/Reference/NSURLSession_class/#//apple_ref/occ/instm/NSURLSession/finishTasksAndInvalidate): `Once a session is invalidated, new tasks cannot be created in the session, but existing tasks continue until completion.`

Comment (shortstuffsushi) [inline Branch-SDK/Branch-SDK/BNCServerInterface.m]: Ah, indeed. Should work then.

**Pull request merged**